### PR TITLE
Modify therapy modal

### PR DESCRIPTION
**Pull request by bprize15**
resolves https://github.com/oncokb/oncokb-pipeline/issues/228

### CODE REVIEW

File: src/main/webapp/app/pages/curation/collapsible/MutationCollapsible.tsx [hunk 1]
```diff
@@ -309,69 +312,96 @@ const MutationCollapsible = ({
...
+                              mutationUuid={mutation.name_uuid}
+                              cancerTypesUuid={tumor.cancerTypes_uuid}
+                              treatmentUuid={treatment.name_uuid}
                             />
                           }
-                          name="evidenceDescription"
+                          action={
+                            <>
+                              <FaEdit
+                                style={{ cursor: 'pointer' }}
```
Comment (zhx828) [inline src/main/webapp/app/pages/curation/collapsible/MutationCollapsible.tsx:331]: Is there a way we can make this style applicable to all FaEdit usage?

File: src/main/webapp/app/pages/curation/CurationPage.tsx [hunk 2]
```diff
@@ -35,6 +35,7 @@ import CurationHistoryTab from 'app/components/tabs/CurationHistoryTab';
 import { FaFilter } from 'react-icons/fa';
 import _ from 'lodash';
 import MutationCollapsible from './collapsible/MutationCollapsible';
+import ModifyTherapyModal from 'app/shared/modal/ModifyTherapyModal';
```
Comment (zhx828) [inline src/main/webapp/app/pages/curation/CurationPage.tsx:38]: not used

File: src/main/webapp/app/shared/modal/modify-therapy-modal.store.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { action, makeObservable, observable } from 'mobx';
+import { DrugSelectOption } from '../select/DrugSelect';
+
+export class ModifyTherapyModalStore {
```
Comment (zhx828) [inline src/main/webapp/app/shared/modal/modify-therapy-modal.store.tsx:4]: I didn't expect this to be a global store, but rather only exists when modal opens. It should be destroyed after closing.

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/pages/curation/collapsible/MutationCollapsible.tsx (hunk 1 above)
Comment (bprize15) [reply src/main/webapp/app/pages/curation/collapsible/MutationCollapsible.tsx:331]: We now have `OncoKBEdit`

File: src/main/webapp/app/shared/modal/modify-therapy-modal.store.tsx (hunk 3 above)
Comment (bprize15) [reply src/main/webapp/app/shared/modal/modify-therapy-modal.store.tsx]: Will readdress after listener refactor

File: src/main/webapp/app/components/icons/OncoKBEdit.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { IconBaseProps } from 'react-icons';
+import { FaEdit } from 'react-icons/fa';
+import { IconContext } from 'react-icons/lib';
+
+export default function OncoKBEdit(props: IconBaseProps) {
```
Comment (zhx828) [inline src/main/webapp/app/components/icons/OncoKBEdit.tsx:6]: This should be moved to shared/icons, and should be named as EditIcon. Same goes to CommentIcon, it should be moved as well.

**Review by zhx828 — CHANGES_REQUESTED** [review]

**Pull request merged**